jsonargparse/core.py

```diff
@@ -589,7 +590,10 @@ def _load_config_parser_mode(
             _jsonnet = import_jsonnet('_load_config_parser_mode')
             cfg_str = _jsonnet.evaluate_snippet(cfg_path, cfg_str, ext_vars=ext_vars, ext_codes=ext_codes)
         try:
-            cfg_dict = load_value(cfg_str)
+            if self.parser_mode == 'jsonnet':
+                cfg_dict = load_value(cfg_str)
+            else:
+                cfg_dict = parse_str(cfg_str, cfg_path = cfg_path)
```

Comment:
I don't think introducing too much non-standard syntax by default is beneficial for long-term project maintenance. You can easily extend it with [custom loader](https://jsonargparse.readthedocs.io/en/stable/#custom-loaders).

Another thing to note is that since jsonargparse can automatically switch current working directory  to the config file path as discussed in https://github.com/omni-us/jsonargparse/issues/122, custom loader does not need to pass in the path to support relative path introdction.